drivers/i2c/i2c_nrf5.c

```diff
@@ -272,6 +272,26 @@ static void i2c_nrf5_isr(void *arg)
 	}
 }
 
+static int i2c_nrf5_recovery(struct device *dev)
```

Comment:
This is not a proper I2C recovery procedure. You should start with setting SCL high, then generate clock pulses (1-0-1) on this line until SDA goes high (at this point the master can control SDA again, so there is no need to generate more pulses). There should be maximum 9 such clock pulses (this is the worst-case scenario, when a slave needs to send 8 bits of data and get the ACK/NAK bit). After getting SDA high you need to generate a stop condition, i.e. make a 0-1 transition on  SDA while keeping SCL high, to finally finish this erroneous transaction. The whole thing is quite nice described for instance in [this application note](http://www.analog.com/media/en/technical-documentation/application-notes/54305147357414AN686_0.pdf).